doc/asciidoc/consumer/index.adoc

```diff
@@ -315,47 +315,96 @@ By default no properties will be attached to the edge nodes but you can specify
 
 === How deal with bad data
 
-The Neo4j Streams Plugin provides a way to re-route all the data that for something reason it wasn't able to ingest to a `Dead Letter Queue`.
-You can active the feature by setting the following property:
+The Neo4j Streams Plugin provides several means to handle processing errors.
 
-```
-streams.sink.dlq=<TOPIC_NAME>
-```
+It can fail fast or log errors with different detail levels.
+Another way is to re-route all the data and errors that for something reason it wasn't able to ingest to a `Dead Letter Queue`.
 
-So given:
+NOTE: It behaves by default like Kafka Connect, see this https://www.confluent.io/blog/kafka-connect-deep-dive-error-handling-dead-letter-queues[blog post^]
 
-```
-streams.sink.dlq=my-dlq-topic
-```
+* fail fast (abort) by default
+* need to configure dead-letter-queue topic to enable
+* need to enable logging explicitly
+* headers and message logging must be enabled explicitly
+
+Config Options
+
+[%autowidth,cols="m,m,a",opts=header]
+|===
+| Name | Value | Note
+| errors.tolerance | none |  fail fast (default!) abort
+| errors.tolerance | all |  all == lenient, silently ignore bad messages
+| errors.log.enable | false/true | log errors (default: false)
+| errors.log.include.messages | false/true | log bad messages too (default: false)
+| errors.deadletterqueue.topic.name | topic-name | dead letter queue topic name, if left off no DLQ, default: not set
+| errors.deadletterqueue.context.header.enable | false/true | enrich messages with metadata headers like exception, timestamp, org. topic, org.part, default:false
+| errors.deadletterqueue.context.header.prefix | prefix-text | common prefix for header entries, e.g. `"__streams.errors."` , default: not set
+| errors.deadletterqueue.topic.replication.factor | 3/1 | replication factor, need to set to 1 for single partition, default:3
+|===
+
+In Kafka Connect you can just configure the above properties.
 
-the Neo4j Streams plugin will re-route all the data to the the `my-dlq-topic` topic.
+For the Neo4j extension you prefix them with `streams.sink.` in the Neo4j configuration.
+
+Example settings:
+
+.Fail Fast, Abort
```

Comment:
What does "abort" mean?  In your code, you throw an exception.  Then what?   Does that exception go to debug.log?  Do you crash neo4j?   Make a note of it.